platform/fabric/core/generic/chaincode/chaincode.go

```diff
@@ -69,3 +70,43 @@ func (c *Chaincode) IsPrivate() bool {
 	// Nothing was found
 	return false
 }
+
+// Version returns the version of this chaincode.
+// It uses discovery to extract this information from the endorsers
+func (c *Chaincode) Version() (string, error) {
+	response, err := NewDiscovery(c).Response()
+	if err != nil {
+		return "", errors.WithMessage(err, "failed to discover")
```

Comment:
I think this error needs more detail here such as failed to discover what ? what was it trying to do